pkg/provision/workspace/pull_secret.go

```diff
@@ -74,6 +75,14 @@ func PullSecrets(clusterAPI sync.ClusterAPI, serviceAccountName, namespace strin
 	}
 	err = clusterAPI.Client.Get(context.TODO(), namespacedName, serviceAccount)
 	if err != nil {
+		if k8sErrors.IsNotFound(err) {
+			// ServiceAccount does not exist, no pull secrets to extract
```

Comment:
Is this because DWO only caches SA's with the devworkspace ID label, so if a user-defined SA is used, the Get request will fail (even though things are working correctly)?

Reply:
Yup, there's no way for us to know if the SA exists or not because we'll only see it if it has the devworkspace_id label. In non-disabled cases, the SA should exist and we should be able to check for pull secrets.